jobs/webcompat-kb/webcompat_kb/main.py

```diff
@@ -78,283 +100,543 @@ def get_urls_from_story(user_story_str):
+                    value, relation_table, job_config=job_config
+                )
+
+                logging.info(f"Writing to `{relation_table}` table")
+
+                try:
+                    job.result()
+                except Exception as e:
+                    print(f"ERROR: {e}")
+                    if job.errors:
+                        for error in job.errors:
+                            logging.error(error)
+
+                table = self.client.get_table(relation_table)
+                logging.info(f"Loaded {table.num_rows} rows into {table}")
+
+    def get_last_import_datetime(self):
+        query = f"""
+                SELECT MAX(run_at) AS last_run_at
+                FROM `{self.bq_dataset_id}.import_runs`
+                WHERE is_history_fetch_completed = TRUE
+            """
+        res = self.client.query(query).result()
+        row = list(res)[0]
+        return row["last_run_at"]
+
+    def fetch_history(self, bug_id, last_import_time=None):
+        if not bug_id:
+            raise ValueError("No bug id provided")
 
-    for bug_id, data in bugs.items():
-        for field_key, items in data.items():
-            fields = relation_config[field_key]["fields"]
+        params = {}
 
-            for row in items:
-                relation_row = {fields[0]["name"]: bug_id, fields[1]["name"]: row}
-                relations[field_key].append(relation_row)
+        if last_import_time:
+            params["new_since"] = last_import_time.strftime("%Y-%m-%dT%H:%M:%SZ")
 
-    return relations
+        headers = {}
 
+        if self.bugzilla_api_key:
+            headers = {"X-Bugzilla-API-Key": self.bugzilla_api_key}
 
-def merge_relations(main_dict, additional_dict):
-    for key, value in additional_dict.items():
-        if key in main_dict:
-            main_dict[key].extend(value)
-        else:
-            main_dict[key] = value
-    return main_dict
+        url = f"{BUGZILLA_API}/bug/{bug_id}/history"
 
+        try:
+            response = requests.get(url, headers=headers, params=params)
+            response.raise_for_status()
+            result = response.json()
+            return result["bugs"][0]
+        except Exception as e:
+            logging.error(f"Error: {e}")
+            self.history_fetch_completed = False
+            return []
 
-def split_bugs(dep_bugs, bug_ids):
-    core_bugs, breakage_bugs = [], []
+    def fetch_bugs_history(self, ids, last_import_time=None):
+        bugs_history = []
 
-    for bug in dep_bugs:
-        if bug["id"] in bug_ids.get("core", []):
-            core_bugs.append(bug)
-        elif bug["id"] in bug_ids.get("breakage", []):
-            breakage_bugs.append(bug)
+        for bug_id in ids:
+            try:
+                logging.info(f"Fetching history from bugzilla for {bug_id}")
+                bug_history = self.fetch_history(bug_id, last_import_time)
+                bugs_history.append(bug_history)
+                time.sleep(2)
```

Comment:
This method would only fetch history for bugs changed since the last run to avoid calling the api unnecessarily.

I couldn't find any information about a rate limit for bugzilla api in the docs. Decided to add this wait in case there are a lot of bugs to fetch histories for at some point (for example, a mass update for all bugs in our component).